pkg/broker/docker_manager.go

```diff
@@ -0,0 +1,114 @@
+package broker
+
+import (
+	"context"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/go-connections/nat"
+	"github.com/pivotal-cf/brokerapi"
+)
+
+type dockerContainerManager struct {
+	client DockerClient
+	logger lager.Logger
+}
+
+func NewDockerContainerManager(logger lager.Logger, client DockerClient) ContainerManager {
+	return dockerContainerManager{
+		client: client,
+		logger: logger.Session("docker-container-manager"),
+	}
+}
+
+func (dc dockerContainerManager) Provision(containerConfig *ContainerConfig) (brokerapi.ProvisionedServiceSpec, error) {
+	ctx := context.Background()
+	_, err := dc.client.ImagePull(ctx, containerConfig.Image, types.ImagePullOptions{})
```

Comment:
/blocker
there's some kind of weird race condition when we test this. may need to do this in a loop or otherwise wait somehow until the image is actually pulled before proceeding.
Additionally, we still need the comment on **why** we'red oing the pull, which makes sense. (it may need to be configurable, but I'm not worried about that now.